base-notebook/start.sh

```diff
@@ -4,12 +4,14 @@
 
 set -e
 
+mkdir -p -m 700 /home/$NB_USER/.local/share/jupyter
```

Comment:
We used to create this directory ahead of time to avoid issues with permissions at runtime. I can't remember the specifics of those problems now, though I vaguely recall they are workarounds. I'm going to try removing the mkdir from the Dockerfile, avoid changing anything in start.sh, and see what happens in the various images.